bids-validator/src/schema/applyRules.ts

```diff
@@ -442,6 +444,43 @@ function evalJsonCheck(
         ]);
       }
     }
+    let originFileKey = ''
+    if (keyName in context.sidecarKeyOrigin) {
+      originFileKey = `${context.sidecarKeyOrigin[keyName]}.${keyName}`
+    }
+    if (context.dataset.sidecarKeyValidated.has(originFileKey)) {
+      return
+    }
+
+    if (keyName in context.sidecar && schema.objects.metadata && keyName in schema.objects.metadata) {
+      // @ts-expect-error
+      const validate = context.dataset.ajv.compile(schema.objects.metadata[keyName])
```

Comment:
Instead of memoizing on the contents of the `objects.metadata.*`, wouldn't it make sense to have something like:

```js
const cache = {}
function getValidator(key, schema) {
    if (!(key in cache)) {
        cache[key] = ajv.compile(schema.objects.metadata[key])
    }
    return cache[key]
}
```

Then:

```suggestion
      const validate = getValidator(keyName, schema)
```